tools/cldr-apps/js-unittest/TestCldrText.js

```diff
@@ -98,11 +98,85 @@
         votes: 3,
       };
       result = cldrText.sub(key, map);
-      resultV = cldrText.test.subVerbose(key, map);
+      resultV = subVerbose(key, map);
       assert(
         result === resultV,
         "[" + result + "]" + " should equal [" + resultV + "]"
       );
     });
   });
+
+  const CLDR_TEXT_DEBUG = false;
+
+  /**
+   * Same as sub(), but more verbose, with subroutines, for unit testing and debugging
+   */
+  function subVerbose(k, map) {
+    const template = cldrText.get(k);
+    if (!template) {
+      console.log("subVerbose: missing template for k = " + k);
+      return "";
+    }
+    if (map instanceof Array) {
+      return fillInBlanksWithArray(template, map);
+    } else if (map instanceof Object) {
+      return fillInBlanksWithObject(template, map);
+    } else {
+      return "";
+    }
+  }
+
+  /**
+   * Get the string that results from filling in blanks in the given template
+   *
+   * @param template a string like "Sorry, your vote for '${1}' could not be submitted: ${0}"
+   * @param map an array like ["too goofy", "🤪"]
+   * @return a string like "Sorry, your vote for '🤪' could not be submitted: too goofy"
+   */
+  function fillInBlanksWithArray(template, map) {
```

Comment:
ok for now, would be good to use an i18n library function eventually.

Reply:
Yes. It's clear that nls/stui.js was meant for i18n -- maybe someday we can use <i>modern</i> dojo or something similar for i18n and actually provide some alternatives to English for the ST UI!

Note that fillInBlanksWithArray is only in the "verbose" version of sub, which is only for unit testing. The production version is only 12 lines -- I was surprised at how small it got:
```
  function sub(k, map) {
    const template = cldrText.get(k);
    if (template) {
      if (map instanceof Array) {
        return template.replace(/\${(\d)}/g, (blank, i) => map[i]);
      }
      if (map instanceof Object) {
        return template.replace(/\${([^}]+)}/g, (blank, i) => map[i]);
      }
    }
    return "";
  }
```